nuit/utils.py

```diff
@@ -13,10 +13,24 @@ def _closure(fn):
         return fn.__closure__
 
 
+def _code(fn):
+    '''
+    Python 2 and Python 3 compatibility function
+    '''
+    try:
+        return fn.func_code
+    except AttributeError:
+        return fn.__code__
+
+
 def has_closure(fn):
     return hasattr(fn, 'func_closure') or hasattr(fn, '__closure__')
 
 
+def has_code(fn):
```

Comment:
I was just looking at https://docs.python.org/3/library/inspect.html#module-inspect, apparently `function` objects are the only things that have `__code__`

I'm not sure if there's anything else that might have these attributes.

The whole section here is about introspecting the view code we're linking to extract any logic about permissions, and run that without running the whole view. We're aiming to improve DRY by not needing to specify menu link visibility when you link to a view, as well as in the view itself.